profiling/src/lib.rs

```diff
@@ -343,7 +343,16 @@ extern "C" fn rinit(r#type: c_int, module_number: c_int) -> ZendResult {
                 .or(environment.profiling_experimental_cpu_enabled.as_deref())
                 .and_then(parse_boolean)
                 .unwrap_or(true);
-            locals.service = environment.service;
+            locals.service = environment.service.or_else(|| {
+                SAPI.get().and_then(|sapi| match sapi {
```

Comment:
So SAPI can reasonably not contain a value. Usage of and_then is nice.

Reply:
Well... sort of. There will always be a SAPI but this a consequence of Rust hating mutable globals. I could remove the optionality by using unsafe bits, but the cost in this case is reasonable enough that I went with safe code. Particularly because there's a graceful failure case here (don't have to worry about returning an error or panic'ing).